modules/dispatch/src/lib.rs

```diff
@@ -297,6 +311,39 @@ impl<T: Config<I>, I: 'static> MessageDispatch<T::AccountId, T::BridgeMessageId>
 			return dispatch_result
 		}
 
+		// Dispatch Ethereum call earlier before pay
+		match T::EthereumCallDispatcher::dispatch(&call, &origin_account) {
```

Comment:
I feel like this is a patch here. 
maybe, generalize the `EthereumCallDispatch` to `CallDispatch`.

or, something like:
replace 
`let origin = RawOrigin::Signed(origin_account).into();` 
with
`let origin = T::OriginBuilder.getOrigin(call, origin_account);`

Reply:
> generalize the EthereumCallDispatch to CallDispatch.

It seems like a rename issue. For dispatch calls besides the Ethereum pallet, we use the original call dispatch in this pallet. The dispatcher I added is only used for the Ethereum pallet dispatch call, so I think `EthereumCallDispatch` is more suitable here.

Reply:
Currently, this one is only used by the Ethereum pallet. I don't see any other pallet that needs it. Rename to `CallDispatch` is wired this pallet already called `pallet-messages-dispatch`, then there is another associated type called `CallDispatch`, After the CallDispatch executes logic, there is another call dispatch.

Reply:
I agree with @wuminzhe , why not use `CallFilter` for validation, and do the duplicate dispatch and cases match  which will be called later.

Why not directly use the spec described in https://github.com/darwinia-network/darwinia-messages-substrate/issues/100 ?